### PR TITLE
Create v24.1 paths and files

### PR DESCRIPTION
**Pull request by mikeCRL**
Modified `_config_cockroachdb.yml` and created new files in `v24.1/`, `_includes/v24.1/`, `images/v24.1/` based on files in their v23.2 counterparts.

### CODE REVIEW

**Comment by mikeCRL** [comment]
@mdlinville Could you please review (to the extent that's possible or necessary for this type of change - may not really be needed) and if you're happy, merge (so it doesn't get out of sync with main while I'm not at my desk)?

File: src/current/_includes/v24.1/sql/show-ranges-output-deprecation-notice.md [hunk 1]
```diff
@@ -1,4 +1,4 @@
-The statement syntax and output documented on this page use the updated `SHOW RANGES` that **will become the default in CockroachDB v23.2**. To enable this syntax and output, set the [cluster setting `sql.show_ranges_deprecated_behavior.enabled`]({% link {{ page.version.version }}/cluster-settings.md %}#setting-sql-show-ranges-deprecated-behavior-enabled)  to `false`:
+The statement syntax and output documented on this page use the updated `SHOW RANGES` that **will become the default in CockroachDB v23.2**. To enable this syntax and output, set the [cluster setting `sql.show_ranges_deprecated_behavior.enabled`]({% link {{ page.version.version }}/cluster-settings.md %})  to `false`:
```
Comment (mdlinville) [inline src/current/_includes/v24.1/sql/show-ranges-output-deprecation-notice.md:1]: I'm confused about the reference to 23.2 here in the 24.1 version of this file. What am I missing?

File: src/current/_data/versions.csv [hunk 2]
```diff
@@ -13,4 +13,4 @@ v22.1,2022-05-24,2023-05-24,2023-11-24,v21.2,release-22.1
 v22.2,2022-12-05,2023-12-05,2024-06-05,v22.1,release-22.2
 v23.1,2023-05-15,2024-05-15,2024-11-15,v22.2,release-23.1
 v23.2,2024-02-05,2025-02-05,2025-08-05,v23.1,release-23.2
-v24.1,N/A,N/A,N/A,v23.2,main
+v24.1,N/A,N/A,N/A,v23.2,master
```
Comment (mdlinville) [inline src/current/_data/versions.csv:16]: Confirming that this is meant to be the `cockroachdb/cockroach` branch, not the docs branch.

**Review by mdlinville — APPROVED** [review]
I'll go ahead and approve. We can scrub the 24.1 files for erroneous mentions of 23.2 later on.

**Pull request merged**